### PR TITLE
OZ Medium severity audit fixes (on develop)

### PR DESCRIPTION
**Pull request by maurelian**
Same changes as in #2749, but applied to develop.

The other PR is for the auditors to do fix review on.

### CODE REVIEW

**Comment by tynes** [comment]
I've added the mergify ignore label so this is not squashed on merge, we should manually merge it once its ready for review

File: packages/contracts-bedrock/contracts/test/OptimismMintableERC20.t.sol [hunk 1]
```diff
@@ -64,4 +66,24 @@ contract OptimismMintableERC20_Test is Bridge_Initializer {
...
         L2Token.burn(alice, 100);
     }
+
+    function test_erc165_supportsInterface() external {
+        // The assertEq calls in this test are comparing the manual calculation of the iface,
+        // with what is returned by the solidity's type().interfaceId, just to be safe.
+        bytes4 iface1 = bytes4(keccak256("supportsInterface(bytes4)"));
+        assertEq(iface1, type(IERC165).interfaceId);
+        assert(L2Token.supportsInterface(iface1));
+        emit log_bytes32(bytes32(iface1));
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/test/OptimismMintableERC20.t.sol:76]: Did you mean to leave these loglines?

**Review by tynes — APPROVED** [review]

File: packages/contracts-bedrock/contracts/L1/OptimismPortal.sol [hunk 2]
```diff
@@ -225,6 +229,12 @@ contract OptimismPortal is ResourceMetering {
             "OptimismPortal: insufficient gas to finalize withdrawal"
         );
 
+        // Disallow withdrawals during reentrancy to this function.
```
[the remark this answers is not in the document]
Comment (maurelian) [reply packages/contracts-bedrock/contracts/L1/OptimismPortal.sol:232]: I probably screwed up a rebase.

File: .changeset/three-pants-punch.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+'@eth-optimism/contracts-bedrock': minor
```
Comment (tynes) [inline .changeset/three-pants-punch.md:2]: I don't think we need to do another minor release but not super opinionated. This package will be deprecated and we will do a major release of `contracts` once we upgrade to bedrock

Comment (maurelian) [reply .changeset/three-pants-punch.md]: changed to patch

File: packages/contracts-bedrock/contracts/test/OptimismMintableERC20.t.sol (hunk 1 above)
Comment (maurelian) [reply packages/contracts-bedrock/contracts/test/OptimismMintableERC20.t.sol]: No, took them out.

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**